### PR TITLE
upgrade: Make --keep-going the default

### PR DESCRIPTION
**Pull request by autoantwort**
Closes [#17922](https://github.com/microsoft/vcpkg/discussions/17922)

### CODE REVIEW

**Review by vkaytsanov — APPROVED** [review]
Ship it

**Comment by strega-nil-ms** [comment]
I like this choice; however, it definitely requires discussion.

File: src/vcpkg/commands.upgrade.cpp [hunk 1]
```diff
@@ -24,11 +24,13 @@ namespace vcpkg::Commands::Upgrade
...
     static constexpr StringLiteral OPTION_NO_DRY_RUN = "no-dry-run";
     static constexpr StringLiteral OPTION_KEEP_GOING = "keep-going";
+    static constexpr StringLiteral OPTION_STOP_AT_FAILURE = "stop-at-failure";
     static constexpr StringLiteral OPTION_ALLOW_UNSUPPORTED_PORT = "allow-unsupported";
 
-    static constexpr std::array<CommandSwitch, 3> INSTALL_SWITCHES = {{
+    static constexpr std::array<CommandSwitch, 4> INSTALL_SWITCHES = {{
         {OPTION_NO_DRY_RUN, "Actually upgrade"},
-        {OPTION_KEEP_GOING, "Continue installing packages on failure"},
+        {OPTION_KEEP_GOING, "The default (deprecated)"},
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.upgrade.cpp:32]:
```suggestion
        {OPTION_KEEP_GOING, "Continue installing packages on failure (default)"},
```

File: src/vcpkg/commands.upgrade.cpp [hunk 2]
```diff
@@ -24,11 +24,13 @@ namespace vcpkg::Commands::Upgrade
...
+        {OPTION_STOP_AT_FAILURE, "Stop installing packages on failure"},
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.upgrade.cpp:33]:
```suggestion
        {OPTION_NO_KEEP_GOING, "Stop installing packages on failure"},
```

File: src/vcpkg/commands.upgrade.cpp [hunk 3]
```diff
@@ -54,8 +56,16 @@ namespace vcpkg::Commands::Upgrade
...
         const ParsedArguments options = args.parse_arguments(COMMAND_STRUCTURE);
 
+        if (Util::Sets::contains(options.switches, OPTION_KEEP_GOING))
+        {
+            print2(Color::warning,
+                   "The option `",
+                   OPTION_KEEP_GOING,
+                   "` is the default now. You don't have to pass this option anymore.");
+        }
+
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.upgrade.cpp:66]:
```suggestion
```

it's not standard practice to warn on passing the "default" for an option.

File: src/vcpkg/commands.upgrade.cpp [hunk 4]
```diff
@@ -54,8 +56,16 @@ namespace vcpkg::Commands::Upgrade
 
         const ParsedArguments options = args.parse_arguments(COMMAND_STRUCTURE);
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.upgrade.cpp:58]: Also, make sure that you check for both `OPTION_KEEP_GOING` and `OPTION_NO_KEEP_GOING` and error if both are passed.

File: src/vcpkg/commands.upgrade.cpp [hunk 5]
```diff
@@ -24,11 +24,13 @@ namespace vcpkg::Commands::Upgrade
 
     static constexpr StringLiteral OPTION_NO_DRY_RUN = "no-dry-run";
     static constexpr StringLiteral OPTION_KEEP_GOING = "keep-going";
+    static constexpr StringLiteral OPTION_STOP_AT_FAILURE = "stop-at-failure";
```
Comment (strega-nil-ms) [inline src/vcpkg/commands.upgrade.cpp:27]:
```suggestion
    static constexpr StringLiteral OPTION_NO_KEEP_GOING = "no-keep-going";
```

this is the standard.

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Thanks, just some minor stuff.

**Review by BillyONeal — APPROVED** [review]

**Review by strega-nil-ms — APPROVED** [review]

**Pull request merged**